packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -0,0 +1,142 @@
+import { useCallback, useRef } from 'react';
+import styled from '@emotion/styled';
+import { useIntl } from 'react-intl';
+import { useShowNotification } from '@commercetools-frontend/actions-global';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import {
+  DOMAINS,
+  NOTIFICATION_KINDS_PAGE,
+} from '@commercetools-frontend/constants';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import CustomPanel from '../custom-panel/custom-panel';
+import messages from './messages';
+
+type TCustomViewIframeMessage = {
+  source: string;
+  destination: string;
+  eventName: string;
+  eventData: Record<string, unknown>;
+};
+
+/*
+  TODO: These types are temporary until we have the proper
+  ones generated from the Settings schema
+*/
+export type TPermissionGroup = {
+  name: string;
+  oAuthScopes: string[];
+};
+
+export type TCustomView = {
+  id: string;
+  defaultLabel: string;
+  labelAllLocales: Record<string, string>;
+  url: string;
+  type: 'custom-panel' | 'custom-tab';
+  typeConfig?: {
+    size?: 'small' | 'large';
+  };
+  locators: string[];
+};
+
+type TCustomViewLoaderProps = {
+  customView: TCustomView;
+  onClose: () => void;
+};
+
+const isIframeReady = (iFrameElementRef: HTMLIFrameElement) => {
+  try {
+    return iFrameElementRef?.contentWindow?.document.readyState === 'complete';
+  } catch {
+    // Trying to access the contentWindow of a cross-origin iFrame will throw an error.
+    // We are not supposed to even get here because the iFrame must use
+    // a URL from our very same domain (the custom view is proxied through our http-proxy service).
```

Comment:
I was hesitant about it.

This should not happen because the type of the Custom View should come from a select input in the configuration view so, if it happens, I think it would be because of a bug in our side, that's why I decided to report it.

But we can also just log it into the console if we prefer to not have this kind of errors in Sentry.

Reply:
Ok, I think it's fine then to report it to Sentry to help us catching such (unlikely) cases.

Reply:
Added here: d790a17fb8197e64f1c44a54c64eec708e911e7e